src/callbacks_step/analysis_dg2d.jl

```diff
@@ -135,3 +150,26 @@ function analyze(::Val{:linf_divb}, du::AbstractArray{<:Any,4}, u, t,
 
   return linf_divb
 end
+
+
+function analyze(::Val{:linf_divb}, du::AbstractArray{<:Any,4}, u, t,
+                 mesh::TreeMesh{2}, equations::IdealGlmMhdMulticomponentEquations2D,
+                 dg::DG, cache)
+  @unpack derivative_matrix, weights = dg.basis
+
+  # integrate over all elements to get the divergence-free condition errors
+  linf_divb = zero(eltype(u))
+  for element in eachelement(dg, cache)
+    for j in eachnode(dg), i in eachnode(dg)
+      divb = zero(eltype(u))
+      for k in eachnode(dg)
+        divb += ( derivative_matrix[i, k] * u[5, k, j, element] +
+        derivative_matrix[j, k] * u[6, i, k, element] )
```

Comment:
```suggestion
        divb += ( derivative_matrix[i, k] * u[5, k, j, element] +
                  derivative_matrix[j, k] * u[6, i, k, element] )
```

Reply:
Instead of copy pasting a lot of this MHD stuff, you could also consider adding a function
`magnetic_flux_densities(u, equation::IdealGlmMhdMulticomponentEquations2D) = SVector(u[5], u[6])`
to all MHD-like systems of equations and then have one implementation for this analysis stuff here. What do you think?

You should also talk to @andrewwinters5000 though, I am not sure if `magnetic_flux_densities` is the best (or even a correct) name for the physical quantity we're looking at here.

Reply:
For now I would suggest to just leave it like that. 
Since I'm playing with the idea to change my multicomponent structure from ("rho_v1", "rho_v2", "rho_v3", "rho_e", "B1", "B2", "B3", "psi", "rho1", "rho2" ..) to (**"rho",** "rho_v1", "rho_v2", "rho_v3", "rho_e", "B1", "B2", "B3", "psi", "rho1", "rho2" ..) in the next PR, this stuff might be obsolete anyway.

Reply:
OK.